scipy/spatial/distance.py

```diff
@@ -1816,8 +1816,8 @@ class MetricInfo:
         aka={'dice'},
         types=['bool'],
         dist_func=dice,
-        cdist_func=CDistMetricWrapper('dice'),
-        pdist_func=PDistMetricWrapper('dice'),
+        cdist_func=_distance_pybind.cdist_dice,
+        pdist_func=_distance_pybind.pdist_dice,
```

Comment:
These boolean metrics have gotten slower which I believe is due to promoting their inputs to float. Maybe we could have a wrapper function that dispatches to the old C code for boolean tensors and the new pybind code for everything else?

Reply:
Without knowing all the context, I believe a minor performance regression is completely acceptable if it allows us to get rid of the old C code. And looking at the benchmarks in the OP, it's gotten _a lot_ faster elsewhere too, making this even more of a worthwhile trade-off IMO.

Reply:
I took the time to fix the bug, but not sure I'll have the bandwidth to dig into this one, can you live with the this tradeoff @peterbell10?